pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -157,75 +68,145 @@ def rydberg_drive(amplitude, phase, detuning, wires):
-    evolution where an additional local drive acting on wires ``[0, 1]`` is added to the Hamiltonian:
+    evolution where an additional local drive that changes in time is acting on wires ``[0, 1]`` is added to the Hamiltonian:
 
-    .. code-block:: python
+    .. code-block:: python3
 
         amplitude_local = lambda p, t: p[0] * jnp.sin(2 * jnp.pi * t) + p[1]
         phase_local = lambda p, t: p * jnp.exp(-0.25 * t)
         detuning_local = jnp.pi / 4
-        H_local = qml.pulse.rydberg_drive(amplitude_local, phase_local, detuning_local, [0, 1])
+        H_local = qml.pulse.drive(amplitude_local, phase_local, detuning_local, [0, 1])
 
-        H = H_i + H_d + H_local
+        H = H_int + H_d + H_local
 
         @jax.jit
         @qml.qnode(dev, interface="jax")
         def circuit_local(params):
             qml.evolve(H)(params, t=[0, 10])
             return qml.expval(qml.PauliZ(0))
 
-    >>> params = [2.4, [1.3, -2.0], -1.5]
+        p_global = 2.4
+        p_amp = [1.3, -2.0]
+        p_phase = 0.5
+        params = (p_global, p_amp, p_phase)
+        circuit_local(params)
+
     >>> circuit_local(params)
-    Array(0.80238028, dtype=float64)
+    Array(0.6126627, dtype=float64)
     >>> jax.grad(circuit_local)(params)
-    [Array(-0.73273312, dtype=float64),
-     [Array(-0.45635171, dtype=float64, weak_type=True),
-      Array(0.76572817, dtype=float64, weak_type=True)],
-     Array(0.01027312, dtype=float64)]
+    (Array(-0.30482848, dtype=float64),
+     [Array(0.27030772, dtype=float64, weak_type=True),
+      Array(-0.55805872, dtype=float64, weak_type=True)],
+     Array(0.27385566, dtype=float64))
+
+    .. details::
+        :title: Theoretical background
+        :href: theory
+
+        Depending on the community and field it is often common to write the driving field Hamiltonian as
+
+        .. math::
+            H = \frac{1}{2} \Omega(t) \sum_{j \in \text{wires}} \left(e^{i (\phi(t) + \nu t)} \sigma^+_j + e^{-i (\phi(t) + \nu t)} \sigma^-_j \right)
+            + \frac{1}{2} \omega_q \sum_{j \in \text{wires}} \sigma^z_j,
+
+        with amplitude :math:`\Omega`, phase :math:`\phi` and drive frequency :math:`\nu` of the electromagnetic field, as well as the qubit frequency :math:`\omega_q`.
+        We can move to the rotating frame of the driving field by applying :math:`U = e^{-i\nu t \sigma^z / 2}` which yields the new Hamiltonian
+
+        .. math::
+            H = \frac{1}{2} \Omega(t) \sum_{j \in \text{wires}} \left(e^{i \phi(t)} \sigma^+_j + e^{-i \phi(t)} \sigma^-_j \right)
+            - \frac{1}{2} (\nu - \omega_q) \sum_{j \in \text{wires}} \sigma^z_j
+
+        We can define :math:`\Delta = \nu - \omega_q` to arrive at the definition above. Note that a potential anharmonicity term, as is common for transmon systems,
+        is unaffected by this transformation. (TODO link transmon anharmonicity constructor)
```

Comment:
As long as we're limited to d=2, I would think the anharmonicity term is meaningless. Do we want to include discussion of this term from the get-go, or is it something we should add to the documentation when we add qutrit/qudit support?

Reply:
I think it makes sense to mention to not confuse users expecting it, but yeah the actual implementation is currently not necessary, good point!